File: MAP/lab2/src/models/CompStmt.java
package models;

import utils.IExeStack;

public class CompStmt implements Statement{

    private Statement first, second;
    public CompStmt(Statement f, Statement s){
        first = f;
        second = s;
    }

    @Override
    public PrgState execute(PrgState state) {
        IExeStack<Statement> stack = state.getStack();
        stack.push(second);
        stack.push(first);

        return state;
    }

    @Override
    public String toString(){
        return "(" + first + second + ")";
    }
}

File: MAP/lab7/lab7/Models/CompStmt.cs
namespace lab7.Models
{
    public class CompStmt : IStatement
    {
        private IStatement first;
        private IStatement second;

        public CompStmt(IStatement first, IStatement second)
        {
            this.first = first;
            this.second = second;
        }
        
        public PrgState execute(PrgState state)
        {
            state.Stack.push(second);
            state.Stack.push(first);
            return state;
        }

        public override string ToString()
        {
            return "[" + first + ";" + second +"]";
        }

       
    }
}